docs/DeveloperGuide.md

```diff
@@ -4,30 +4,274 @@
+Internally, the `list` operation does not modify any data but simply retrieves and displays the information from `EventList`.
+
+The `ListCommand` class performs the following key operations:
+
+* ListCommand#execute() — Generates a formatted message displaying all events in the list.
+
+These operations are accessible through the `Command` and can be invoked when the list command is entered by the user.
+
+#### Feature Implementation
+
+Given below is an example usage scenario and the behavior of the list feature at each step:
+
+1. User Command Input:
+The user enters the command `list` to view all scheduled events.
+
+2. Command Recognition:
+The `COMMAND_WORD` is set to "list", enabling the system to recognize the command input and invoke `ListCommand`.
+
+3. Execution of ListCommand#execute():
+The `execute()` method retrieves each event from `eventList` and appends it to a formatted output message.
+It uses the `String.format` method with `LIST_MESSAGE` to include the total number of events in the message header.
+Events are appended to `outputMessage` with numbered formatting for readability.
+
+4. Output Generation:
+The method stores the generated `outputMessage` in `this.message`, ready for display.
+
+### Add feature
+
+The `add` feature allow users to add events/participants based on relevant flags.
+It is implemented in the `AddCommand` class which extends the base `Command` class and parse through the command to retrieve information based off flags.
+
+The above operation is implemented as `AddCommand#execute()`. This overrides the `Command#execute()` operation in `Command`,
+and is invoked when the latter operation is called.
+
+The `AddCommand` handles two primary functions:
+
+1. **Add an Event:** When provided with event details, including name, time, venue and priority, it creates and stores a new event in the event list.
+2. **Add a Participant:** When provided with participant information, including name, contact number, and email, it attempts to add the participant to an existing event.
+
+#### Feature Implementation
+
+Given below is an example usage scenario for the `add` mechanism, and how it behaves at each step.
+
+1. The user enters the command `add` followed by `-e` or `-p` to indicate adding an event or participant.
+2. This step is determined by our `Parser` which parses through the user input to determine if it is adding a participant or event
+3. Based on the parsed input, `AddCommand` executes one of the following actions:
+   + **Add Event:** Creates a new event in `EventList` with the provided event details
+   + **Add Participant:** Locates the event in `EventList` and adds the participant to it
+4. If a duplicate event is found, `DuplicateDataException` is thrown.
+
+The interactions between components during the execution of the `add` command are show in the **Sequence Diagram** below:
+
+**Add Event**
+
+<img src = "images/AddEventSequenceDiagram.png">
```

Comment:
The `AddEvent` sequence diagram differs very greatly from the one shown near the start of the DG under Design/Interaction between components. 
1. The flags used are different, starting with the dashes/hyphens and the backslash. Consider improving the consistency of the flags used? It should be gathered from the codebase.
2.  The constructor of the `AddCommand` is invoked wrongly. Perhaps refer to the Interaction between components diagram to see how it might be done? Do review the constructor invocation for the downstream classes/objects.
3. Assuming that these are concrete instances, the colon should be present before the class name.
4. There is mention of providing `Participant` information. What is that and what does it look like? And how would I associate a `Participant` to an `Event`?
5. Do consider dedicating a section to explaining `Participant` if it is an important component of the program.
6. Consider introducing the various `alt` blocks that will describe the flow in the different cases of execution?